homeassistant/components/utility_meter/select.py

```diff
@@ -82,12 +84,16 @@ class TariffSelect(SelectEntity, RestoreEntity):
     def __init__(
         self,
         name,
-        tariffs,
-        unique_id,
+        tariffs: list[str],
+        *,
+        yaml_slug: str | None = None,
+        unique_id: str | None = None,
         device_info: DeviceInfo | None = None,
     ) -> None:
         """Initialize a tariff selector."""
         self._attr_name = name
+        if yaml_slug:  # Backwards compatibility with YAML configuration entries
```

Comment:
Could consider here to only do this only if we don't have a unique id
`if not unique_id and yaml_slug:`

Reply:
that would be a breaking change

Reply:
No, if it has a unique id it's stored in the entity registry and will grab it from there on startup.
But we can leave this as is.

Reply:
yet there are configurations predating the introduction of unique_id that would break - including one of the tests